dvc/repo/experiments/queue/local.py

```diff
@@ -287,25 +245,14 @@ def kill(self, revs: Collection[str]) -> None:
         for queue_entry in to_kill:
             self.proc.kill(queue_entry.stash_rev)
 
-    def _shutdown_handler(self, task_id: str = None, **kwargs):
-        if task_id in self._shutdown_task_ids:
-            self._shutdown_task_ids.remove(task_id)
-        if not self._shutdown_task_ids:
-            self.celery.control.shutdown()
-
     def shutdown(self, kill: bool = False):
-        from celery.signals import task_postrun
-
-        tasks = list(self._iter_active_tasks())
-        if tasks:
-            for task_id, _ in tasks:
-                self._shutdown_task_ids.add(task_id)
-            task_postrun.connect()(self._shutdown_handler)
-            if kill:
-                for _, task_entry in tasks:
+        self.celery.control.shutdown()
+        if kill:
+            for _, task_entry in self._iter_active_tasks():
+                try:
                     self.proc.kill(task_entry.stash_rev)
-        else:
-            self.celery.control.shutdown()
+                except ProcessLookupError:
+                    continue
```

Comment:
Since exp task is atomic, we can rely on the default celery `shutdown()` behavior to shut down worker after the currently running task has completed.